cli/src/cmd_auth.rs

```diff
@@ -401,243 +420,72 @@ impl CmdAuthLogout {
-
-        // Include login information from environment variables if set
-        if let Some(h) = std::env::var_os("OXIDE_HOST") {
-            let env_token = match std::env::var_os("OXIDE_TOKEN") {
-                Some(t) => t.into_string().unwrap(),
-                None => String::from(""),
-            };
-            let info = Host {
-                token: env_token,
-                user: String::from(""),
-                default: false,
-            };
-
-            host_list.insert(h.into_string().unwrap(), info);
-        }
-
-        if host_list.is_empty() {
-            return Err(anyhow!("You are not logged into any Oxide hosts."));
-        }
-
-        // Return a single result if the --host flag has been set
-        if let Some(url) = &self.host {
-            if host_list.contains_key(url.as_ref()) {
-                host_list.retain(|k, _| k == url.as_str())
-            } else {
-                return Err(anyhow!(
-                    "Host {} Not found in hosts.toml file or environment variables.",
-                    url.as_str()
-                ));
-            }
-        }
+    pub async fn status(&self, ctx: &Context) -> Result<()> {
+        // For backward compatibility, we'll check OXIDE_HOST and OXIDE_TOKEN
+        // first.
+        if let (Ok(host_env), Ok(token_env)) =
+            (std::env::var("OXIDE_HOST"), std::env::var("OXIDE_TOKEN"))
+        {
+            let client = Client::new_authenticated_config(
+                &ClientConfig::default().with_host_and_token(&host_env, &token_env),
+            )
+            .expect("client authentication from host/token failed");
 
-        for (host, info) in host_list.iter() {
-            // Construct a client with each host/token combination
-            let client = make_client(host, info.token.clone(), ctx.config());
-
-            let result = client.current_user_view().send().await;
-            let user = match result {
-                Ok(usr) => usr,
-                Err(_) => {
-                    // TODO we need to examine the error
-                    status_info.insert(
-                        host.to_string(),
-                        vec![String::from(
-                            "Not authenticated. Host/token combination invalid",
-                        )],
-                    );
-                    continue;
+            match client.current_user_view().send().await {
+                Ok(user) => {
+                    log::debug!("success response for {} (env): {:?}", host_env, user);
+                    println!("Logged in to {} as {}", host_env, user.id)
+                }
+                Err(e) => {
+                    log::debug!("error response for {} (env): {}", host_env, e);
+                    println!("{}: {}", host_env, Self::error_msg(&e))
                 }
             };
+        } else {
+            for (profile_name, profile_info) in &ctx.cred_file().profile {
```

Comment:
It'd be nice to have a way to check the status of a single profile. WDYT?

Reply:
Sure! What do you propose? Can we do it as a follow-on?